ui/src/data_explorer/actions/view/index.js

```diff
@@ -4,10 +4,10 @@ import {getQueryConfig} from 'shared/apis'
 
 import {errorThrown} from 'shared/actions/errors'
 
-export const addQuery = () => ({
+export const addQuery = (queryID = uuid.v4()) => ({
```

Comment:
cool approach. tested to make sure this will generate a new UUID each time the function is invoked, and it does.